app/controllers/declaration/ProcedureCodesController.scala

```diff
@@ -150,13 +150,15 @@ class ProcedureCodesController @Inject()(
         .transform(removeWarehouseIdentificationForCode)
     }
 
-    def updatedModel(model: ExportsDeclaration): ExportsDeclaration =
+    def updatedModel(model: ExportsDeclaration): ExportsDeclaration = {
+      val updatedModel = model.updatedItem(itemId, item => item.copy(procedureCodes = Some(updatedProcedureCodes)))
       updatedProcedureCodes.procedureCode match {
         case Some(code) =>
-          clearDataForProcedureCode(code, itemId, model).updatedItem(itemId, item => item.copy(procedureCodes = Some(updatedProcedureCodes)))
+          clearDataForProcedureCode(code, itemId, updatedModel)
         case _ =>
-          model.updatedItem(itemId, item => item.copy(procedureCodes = Some(updatedProcedureCodes)))
+          updatedModel
```

Comment:
Ah so the whole change is around updating model before clearing data. Previously we first clean and after that updated.